Sources/AuthFoundation/Token Management/Internal/KeychainTokenStorage.swift

```diff
@@ -87,69 +96,74 @@ class KeychainTokenStorage: TokenStorage {
         }
     }
     
-    func replace(token id: String, with token: Token) throws {
-        let oldResult = try Keychain
+    func replace(token id: String, with token: Token, security: [Credential.Security]?) throws {
+        guard let oldResult = try Keychain
             .Search(account: id,
                     service: KeychainTokenStorage.serviceName)
-            .get()
-        
-        let oldToken = try self.token(with: oldResult)
-        try oldResult.delete()
+                .list()
+                .first
+        else {
+            throw TokenError.cannotReplaceToken
+        }
         
         token.id = id
         
         let data = try encoder.encode(token)
-        try Keychain
-            .Item(account: id,
-                  service: KeychainTokenStorage.serviceName,
-                  accessibility: .unlocked,
-                  value: data)
-            .save()
+        let accessibility = security?.accessibility ?? .afterFirstUnlock
+        let accessGroup = security?.accessGroup
+        let accessControl = try security?.createAccessControl(accessibility: accessibility)
+        let newItem = Keychain.Item(account: id,
+                                    service: KeychainTokenStorage.serviceName,
+                                    accessibility: accessibility,
+                                    accessGroup: accessGroup,
+                                    accessControl: accessControl,
+                                    synchronizable: accessibility.isSynchronizable,
+                                    label: nil,
+                                    description: nil,
+                                    generic: nil,
+                                    value: data)
         
-        delegate?.token(storage: self, replaced: id, from: oldToken, to: token)
+        try? oldResult.delete()
+        try newItem.save(authenticationContext: security?.context)
```

Comment:
Why we don't use `update` API here?

Reply:
Update has some limitations, in particular it doesn't let you return a reference of the newly created keychain item. So for cases such as this, a delete/remove is often safer. 

I plan to use `update` in the setMetadata method however.

Reply:
You basically already did the GET of oldResult so you have all required attributes to copy into the new Item. I still like the GET/UPDATE approach since it is only 2 operations comparing to GET/DELETE/ADD. Data safety aspect is also important

Reply:
You're right, in this particular use-case you're totally correct, this is a prime candidate for Update.  I'll make that change.